cmd/influxd/launcher/cmd.go

```diff
@@ -70,24 +70,34 @@ func cmdRunE(ctx context.Context, o *InfluxdOpts) func() error {
 	return func() error {
 		fluxinit.FluxInit()
 
-		// exit with SIGINT and SIGTERM
-		ctx = signals.WithStandardSignals(ctx)
-
 		l := NewLauncher()
-		if err := l.run(ctx, o); err != nil {
+		if err := l.runUntilDone(ctx, o); err != nil {
 			return err
 		}
-		<-ctx.Done()
-
-		// Attempt clean shutdown.
-		ctx, cancel := context.WithTimeout(ctx, 2*time.Second)
-		defer cancel()
-		l.Shutdown(ctx)
+		l.gracefulShutdown(ctx)
 
 		return nil
 	}
 }
 
+// runUntilDone starts the launcher and waits for it to exit on SIGINT or SIGTERM.
+func (l *Launcher) runUntilDone(ctx context.Context, o *InfluxdOpts) error {
+	ctx = signals.WithStandardSignals(ctx)
```

Comment:
Split these into separate methods to make it harder for the run & shutdown contexts to collide with one another